include/dgl/base_heterograph.h

```diff
@@ -381,6 +398,62 @@ struct HeteroSubgraph : public runtime::Object {
   DGL_DECLARE_OBJECT_TYPE_INFO(HeteroSubgraph, runtime::Object);
 };
 
+/*! \brief The flattened heterograph */
+struct FlattenedHeteroGraph : public runtime::Object {
```

Comment:
My general concern is can we use existing python data structure to realize this? For example, induced_eid and induced_nid can be stored as node/edge features. Induced srctype and dsttype can just be the `ntypes` and `etypes` argument to the `DGLHeterograph` initializer and can be stored together natively with the returned graph.

Reply:
I thought when we do wildcard slicing, we need to combine the different source/destination/edge types into one source/destination/edge type and return a unitgraph.  So to keep track of the original node/edge type/ID in the heterograph, we cannot reuse the `ntypes` and `etypes` argument since they actually only contain one type.
Regarding to storing `induced_foo` as features, I'm not sure how to decide the column names to store the node/edge type/IDs.  Should we reserve some names?